src/module/actions/saveAction.js

```diff
@@ -1,37 +1,59 @@
 import { diff } from 'deep-object-diff'
 import { processResponseData } from '../helpers/processResponseData'
 import { getExpectedResponse } from '../helpers/getExpectedResponse'
+import { saveOptions as applySaveOptions } from './options/saveOptions'
 
 /**
  * Update an existing resource
  *
  * @param {ResourcefulApi} api
+ * @param {Boolean} isCollection
  * @param {String} moduleName
+ * @param {Object} defaultQuery
  */
-export function saveAction (api, isCollection, moduleName) {
+export function saveAction (api, isCollection, moduleName, defaultQuery = {}) {
   return new Proxy(() => {}, {
-    apply (target, thisArg, [vuexFns, id]) {
+    apply (target, thisArg, [vuexFns, params]) {
+      const id = (typeof params === 'object') ? params.id : params
+      const options = Object.prototype.hasOwnProperty.call(params, 'options') ? params.options : null
+
       if (typeof id === 'undefined') {
         throw new Error('You must pass an object id to this action')
       }
 
-      const currentItemState = JSON.parse(JSON.stringify((isCollection)
-        ? thisArg.state[moduleName].items[id]
-        : thisArg.state[moduleName].item))
+      let currentItemState = null
+      let initialItemState = null
+      if (Object.prototype.hasOwnProperty.call(defaultQuery, 'group')) {
+        currentItemState = JSON.parse(JSON.stringify((isCollection)
+          ? thisArg.state[moduleName][defaultQuery.group].items[id]
+          : thisArg.state[moduleName][defaultQuery.group].item))
+
+        initialItemState = JSON.parse(JSON.stringify((isCollection)
+          ? thisArg.state[moduleName][defaultQuery.group].initial[id]
+          : thisArg.state[moduleName][defaultQuery.group].initial))
+      } else {
+        currentItemState = JSON.parse(JSON.stringify((isCollection)
+          ? thisArg.state[moduleName].items[id]
+          : thisArg.state[moduleName].item))
 
-      const initialItemState = JSON.parse(JSON.stringify((isCollection)
-        ? thisArg.state[moduleName].initial[id]
-        : thisArg.state[moduleName].initial))
+        initialItemState = JSON.parse(JSON.stringify((isCollection)
+          ? thisArg.state[moduleName].initial[id]
+          : thisArg.state[moduleName].initial))
+      }
```

Comment:
While probably not something to be changed in this PR, this is getting out of hand.

Reply:
indeed. Will try to think about something more elegant

Reply:
But this Task is not for this Feature-Branch